src/goto-instrument/contracts/contracts.cpp

```diff
@@ -911,6 +911,8 @@ void code_contractst::instrument_call_statement(
   }
   else if(callee_name == "free")
   {
+    source_locationt location_no_checks = instruction_it->source_location();
+    disable_pointer_checks(location_no_checks);
```

Comment:
We could add a comment here again explaining "why" is safe to always remove pointer checks in this location.